packages/ws-server/src/webSocketServer.ts

```diff
@@ -44,11 +44,22 @@ const relay: Relay = new RelayImpl(logger, register);
 const app = websockify(new Koa(), {});
 const LOGGER_PREFIX = 'WebSocket:';
 
+const MAX_CONNECTIONS = parseInt(process.env.CONNECTION_LIMIT || '10');
+let connectedClients = 0;
+
 app.ws.use((ctx) => {
     ctx.websocket.id = relay.subs().generateId();
     logger.info(`${LOGGER_PREFIX} new connection ${ctx.websocket.id}`);
 
+    if (connectedClients >= MAX_CONNECTIONS) {
+        ctx.websocket.send("Connections exceeded");
```

Comment:
could message be changed to something like: "Max concurrent connections limit exceeded, please try again later"